sessions/debugging.qmd

```diff
@@ -1,25 +1,126 @@
 # Checking correctness of code: Debugging
 
--   Using basic debugging strategies ('breakpoints', simple `print()` or
-    `stop()` calls, `debug()`, `browser()`)
--   Can take some from the intermediate course (I think that's where it
-    is)
-
 {{< include ../includes/_wip.qmd >}}
 
 ```{r setup}
 #| include: false
 ```
 
-<!-- TODO: Add an introduction here -->
+**Introduction**
+
+Debugging is one of activities that seems really scary and difficult,
+but once you try it and use it, is not nearly as intimidating as it
+seemed. To debug, which means to find and fix problems in your code,
+there are several ways, e.g. simply adding 'breakpoints', `print()` or
+`stop()` calls to your code, or by using the interactive debugger. All
+these methods help you stop and/or inspect the code during its
+execution, which is very helpful if you suspect there is a problem in
+the code.
 
 **Session objectives**:
 
--   Enter the interactive debugger and use it to explore why unexpected
-    behavior is occurring
+-   Learn how to use RStudio's interactive debugger to explore
+    unexpected behavior in your code.
```

Comment:
Amazing start 🎉 🍻 see my comments on the other PRs (#13 and #15) for how I started revising the learning objective/outcome. In general, you want to avoid using the word "learn" in the learning objective, and instead use more explicit and specific language. For instance, see https://tips.uark.edu/using-blooms-taxonomy/ for verbs to use.

In general, in these courses, we focus on "remembering", "understanding", "applying", and sometimes "creating" levels of learning. 

So, in this case, the learning objective could be rewritten as:

```suggestion
-   Identify where the debugger tool is in RStudio and apply it to debug functions that cause issues.
```